lib/internal/process.js

```diff
@@ -22,6 +23,46 @@ const assert = process.assert = function(x, msg) {
 };
 
 
+// Set up the process.cpuUsage() function.
+function setup_cpuUsage() {
+  // Get the native function, which will be replaced with a JS version.
+  const _cpuUsage = process.cpuUsage;
+
+  // Create the argument array that will be passed to the native function.
+  const cpuValues = new Float64Array(2);
+
+  // Replace the native function with the JS version that calls the native
+  // function.
+  process.cpuUsage = function cpuUsage(prevValue) {
+    // Call the native function to get the current values
+    if (_cpuUsage(cpuValues) !== 0) {
+      throw new Error('unable to obtain CPU usage time');
+    }
+
+    // If no previous value passed in, return current value.
+    if (prevValue == null) return {
+      user: cpuValues[0],
+      system: cpuValues[1]
+    };
+
+    // If a previous value was passed in, ensure it has the correct shape.
+    if (!Number.isFinite(prevValue.user)) {
```

Comment:
May want to change this to `Number.isInteger()`, and to be even safer may use the following:
```js
if (!Number.isInteger(prevValue.user) ||
    prevValue.user > Number.MAX_SAFE_INTEGER ||
    prevValue.user < 0) {
  throw new TypeError('previous value "user" is not a safe integer');
}
```
Also noticed how I changed `Error` below to `TypeError`.